public_routing.tf

```diff
@@ -72,5 +72,9 @@ resource "aws_route" "internet_gateway" {
     create = "10m"
   }
 
-  depends_on = [var.module_depends_on]
+  depends_on = [
+    var.module_depends_on,
+    aws_internet_gateway.internet_gateway,
+    aws_route_table.public
```

Comment:
why? they are already known as dependencies by terraform as they are used in the resource